deployment/ccip/changeset/cs_home_chain.go

```diff
@@ -322,3 +326,99 @@ func addNodes(
 	_, err = chain.Confirm(tx)
 	return err
 }
+
+type RemoveDONsConfig struct {
+	HomeChainSel uint64
+	DonIDs       []uint32
+	MCMS         *MCMSConfig
+}
+
+func (c RemoveDONsConfig) Validate(homeChain CCIPChainState) error {
+	if c.HomeChainSel == 0 {
+		return fmt.Errorf("home chain selector must be set")
+	}
+	if len(c.DonIDs) == 0 {
+		return fmt.Errorf("don ids must be set")
+	}
+	// Cap reg must exist
+	if homeChain.CapabilityRegistry == nil {
+		return fmt.Errorf("cap reg does not exist")
+	}
+	if homeChain.CCIPHome == nil {
+		return fmt.Errorf("ccip home does not exist")
+	}
+	// DON ids must exist
+	dons, err := homeChain.CapabilityRegistry.GetDONs(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get dons: %w", err)
+	}
+	for _, donID := range c.DonIDs {
+		exists := false
+		for _, don := range dons {
+			if don.Id == donID {
+				exists = true
+				break
+			}
+		}
+		if !exists {
+			return fmt.Errorf("don id %d does not exist", donID)
+		}
+	}
+	return nil
+}
+
+// RemoveDONs removes DONs from the CapabilitiesRegistry contract.
+// TODO: Could likely be moved to common, but needs
+// a common state struct first.
+func RemoveDONs(e deployment.Environment, cfg RemoveDONsConfig) (deployment.ChangesetOutput, error) {
+	state, err := LoadOnchainState(e)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	homeChain := e.Chains[cfg.HomeChainSel]
+	txOpts := homeChain.DeployerKey
+	if cfg.MCMS != nil {
+		// Remove all DONs
+		txOpts = deployment.SimTransactOpts()
+	}
+	homeChainState := state.Chains[cfg.HomeChainSel]
+	tx, err := homeChainState.CapabilityRegistry.RemoveDONs(txOpts, cfg.DonIDs)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	if cfg.MCMS == nil {
+		_, err = homeChain.Confirm(tx)
+		if err != nil {
+			return deployment.ChangesetOutput{}, err
+		}
+		return deployment.ChangesetOutput{}, nil
```

Comment:
If err == nil it seems like we return early here?

Reply:
a positive log will be assuring